laspy/header.py

```diff
@@ -399,6 +406,17 @@ def add_extra_dims(self, params: List[ExtraBytesParams]) -> None:
     def add_extra_dim(self, params: ExtraBytesParams):
         self.add_extra_dims([params])
 
+    def add_crs(self, crs: pyproj.CRS) -> None:
+        if self.point_format.id >= 6:
```

Comment:
This is a bit unclear to me. The las spec suggests we should check point format for backwards compatibility:

> GeoTIFF is being replaced by Well Known Text (WKT) as the required Coordinate Reference
> System (CRS) representation for the new point types (6-10) introduced by LAS 1.4.
> 
> GeoTIFF is maintained for legacy reasons for point types 0-5.
> 
> A “WKT” bit has been added to the Global Encoding flag in the Public Header Block. If this bit is
> set, the CRS for the file will be located in the WKT (Extended) Variable Length Records (EVLR,
> VLR).
> 
> A file writer who desires to maintain backward compatibility with legacy LAS for point types 0-5
> must add a GeoTIFF VLR to represent the CRS for the file and ensure that the WKT bit is false. 

But it sounds like this is a choice left to the writer? Do you think it should be backwards compatible with older las point types? Happy to change it, just wanted to double check first.

Reply:
Ok, so what I propose, since it seems that we have the choice is to add an optional parameter `keep_compatibility` (or something akin to that) defaulting to True.

And in the function if `keep_comaptibility is True` we check against point format otherwise we check against the version.

Also we have to set the wkt bit of the global encoding thing `self.global_encoding = True/ False `

What do you think about that ?